crates/bevy_geometry/src/lib.rs

```diff
@@ -0,0 +1,356 @@
+use bevy_math::*;
+use bevy_reflect::Reflect;
+use std::error::Error;
+use std::fmt;
+
+pub trait Primitive3d {
+    /// Returns true if this primitive is entirely on the outside (in the normal direction) of the
+    /// supplied plane.
+    fn outside_plane(&self, plane: Plane) -> bool;
+}
+
+#[derive(Debug, Clone)]
+pub enum PrimitiveError {
+    MinGreaterThanMax,
+    NonPositiveExtents,
+}
+impl Error for PrimitiveError {}
```

Comment:
https://github.com/bevyengine/bevy/search?q=thiserror

I see lots of use of `thiserror` in other bevy crates FYI

Reply:
Thanks for the heads up. For some reason I didn't think this was the case.